File: FindLastNegative.py
# Find Last Negative Number in List
# - Given a list of integers nums, return the last negative number.
# - Time: O(n), Space: O(1)

class Solution:
    def findLastNegative(self, nums: list[int]) -> int:
        last_negative = None
        for num in nums:
            if num < 0:
                last_negative = num
        return last_negative

File: FindLastNegative.cs
// Find Last Negative Number in List
// - Given a list of integers nums, return the last negative number.
// - Time: O(n), Space: O(1)

using System;
using System.Collections.Generic;

class Solution
{
    public int? FindLastNegative(List<int> nums)
    {
        int? lastNegative = null;

        foreach (var num in nums)
        {
            if (num < 0)
            {
                lastNegative = num;
            }
        }

        return lastNegative;
    }
}
